code/datums/components/hide_weather_planes.dm

```diff
@@ -0,0 +1,116 @@
+	var/list/datum/weather/active_weather = list()
+	var/list/atom/movable/screen/plane_master/plane_masters = list()
+
+/datum/component/hide_weather_planes/Initialize(atom/movable/screen/plane_master/care_about)
+	if(!istype(parent, /datum/plane_master_group))
+		return COMPONENT_INCOMPATIBLE
+	var/datum/plane_master_group/home = parent
+	plane_masters += care_about
+	RegisterSignal(care_about, COMSIG_QDELETING, PROC_REF(plane_master_deleted))
+
+	var/list/starting_signals = list()
+	var/list/ending_signals = list()
+	for(var/datum/weather/weather_type as anything in typesof(/datum/weather))
+		starting_signals += COMSIG_WEATHER_TELEGRAPH(weather_type)
+		ending_signals += COMSIG_WEATHER_END(weather_type)
+
+	RegisterSignals(SSdcs, starting_signals, PROC_REF(weather_started))
+	RegisterSignals(SSdcs, ending_signals, PROC_REF(weather_finished))
+
+	if(home.our_hud)
+		attach_hud(home.our_hud)
+	else
+		RegisterSignal(home, COMSIG_GROUP_HUD_CHANGED, PROC_REF(new_hud_attached))
+
+/datum/component/hide_weather_planes/Destroy(force)
+	hide_planes()
+	active_weather = null
+	plane_masters = null
+	return ..()
+
+/datum/component/hide_weather_planes/InheritComponent(datum/component/new_comp, i_am_original, atom/movable/screen/plane_master/care_about)
+	if(!i_am_original)
+		return
+	plane_masters += care_about
+	RegisterSignal(care_about, COMSIG_QDELETING, PROC_REF(plane_master_deleted))
+	if(length(active_weather))
+		care_about.enable_alpha()
+	else
+		care_about.disable_alpha()
+
+/datum/component/hide_weather_planes/proc/new_hud_attached(datum/source, datum/hud/new_hud)
+	SIGNAL_HANDLER
+	if(new_hud)
+		attach_hud(new_hud)
+
+/datum/component/hide_weather_planes/proc/attach_hud(datum/hud/new_hud)
+	RegisterSignal(new_hud, COMSIG_HUD_Z_CHANGED, PROC_REF(z_changed))
+	var/mob/eye = new_hud?.mymob?.client?.eye
+	var/turf/eye_location = get_turf(eye)
+	z_changed(new_hud, eye_location?.z)
+
+/datum/component/hide_weather_planes/proc/plane_master_deleted(atom/movable/screen/plane_master/source)
+	SIGNAL_HANDLER
+	plane_masters -= source
+
+/datum/component/hide_weather_planes/proc/display_planes()
+	var/datum/plane_master_group/home = parent
+	var/mob/our_lad = home.our_hud?.mymob
+	var/our_offset = GET_TURF_PLANE_OFFSET(our_lad)
+	for(var/atom/movable/screen/plane_master/weather_concious as anything in plane_masters)
+		//We need to make sure that planes above us are hidden, but below us are visible
+		if(!weather_concious.alpha_enabled && weather_concious.offset >= our_offset)
+			weather_concious.enable_alpha()
+
+/datum/component/hide_weather_planes/proc/hide_planes()
+	for(var/atom/movable/screen/plane_master/weather_concious as anything in plane_masters)
+		weather_concious.disable_alpha()
+
+/datum/component/hide_weather_planes/proc/z_changed(datum/source, new_z)
+	SIGNAL_HANDLER
+	/**
+	 * We hide all impacted planes on z change first because weather planes on lower offsets will show through the game world
```

Comment:
what do you mean by this line? can you post photos?

Reply:
turns out it wasn't needed to be done like that